toolchain/mfc/case_validator.py

```
@@ -1288,6 +1288,32 @@ def check_chemistry(self):
         is provided. No static validation is performed here - chemistry will fail at
         runtime if misconfigured.
         """
+        # Fetch global chemistry and diffusion flags
+        chemistry = self.get("chemistry", "F") == "T"
+        diffusion = self.get("chem_params%diffusion", "F") == "T"
+
+        # Define what constitutes a wall (-15 for slip, -16 for no-slip)
+        wall_bcs = [-15, -16]
+
+        for dir in ["x", "y", "z"]:
```

Ruff (0.15.9)

[error] Variable `dir` is shadowing a Python builtin

(A001)